src/features/organizations/components/OrganizationSwitcher/useOrgSwitcher.ts

```diff
@@ -0,0 +1,99 @@
+import Fuse from 'fuse.js';
+import { RootState } from 'core/store';
+import { TreeItemData } from 'features/organizations/types';
+import useLocalStorage from 'zui/hooks/useLocalStorage';
+import { useMemo } from 'react';
+import { useSelector } from 'react-redux';
+
+function useOrgSwitcher(orgId: number, searchString: string) {
+  function notEmpty<TValue>(value: TValue | null | undefined): value is TValue {
+    return value !== null && value !== undefined;
+  }
```

Comment:
This is obviously a very pretty TS implementation, and if I understand it correctly the [type predicate](https://www.geeksforgeeks.org/what-is-type-predicates-in-typescript/) makes it so that when this function is used in a `filter`, the resulting type is `TValue[]` without null or undefined. That's great!

I think it should be a reusable utility! 🛠️

Reply:
I did not make this as part of this work, I just moved it into the hook, it was made in earlier work on the sidebar. But I'll put it in a utility function!

Reply:
Aah it was @rebecarubio! Credit where credit is due. 🌴